packages/packagers/js/src/ScopeHoistingPackager.js

```diff
@@ -538,6 +544,14 @@ ${code}
       this.needsPrelude = true;
     }
 
+    if (this.bundle.env.isNode() && asset.meta.has_node_replacements) {
+      const relPath = normalizeSeparators(
+        path.relative(this.bundle.target.distDir, path.dirname(asset.filePath)),
+      );
+      code = code.replace('$parcel$dirnameReplace', relPath);
+      code = code.replace('$parcel$filenameReplace', relPath);
+    }
```

Comment:
This should happen further up in this function (e.g. near `if (code.includes('$parcel$global'))`) because there are recursive calls inside `buildAsset` where it inlines dependencies. Right now, `code` might contain the code for multiple assets and then `asset.filePath` is wrong.